### ISSUE TITLE
Tuple size when multiply by a scalar is unknown while it can be guessed.

### ISSUE THREAD

**Issue by AiroPi** [microsoft/pyright#5994]
**Describe the bug**
When multiplying a tuple by a scalar, pyright guesses the type as a tuple with an indeterminate size.
But when the scalar is a Litteral, the size can be known. 

**Code or Screenshots**
```py
def a(b: tuple[int, int]):
    pass

a((1,) * 2)
```

```
Argument of type "tuple[Literal[1], ...]" cannot be assigned to parameter "b" of type "tuple[int, int]" in function "a"
  "tuple[Literal[1], ...]" is incompatible with "tuple[int, int]"
    Tuple size mismatch; expected 2 but received indeterminate
```

**VS Code extension or command-line**
Pyright from Pylance (v2023.9.21) in VSCode Insiders.

**Comment by erictraut** [comment]
The type result for a multiply operation comes from the `tuple` class definition in the typeshed stubs.

```python
class tuple(Sequence[_T_co], Generic[_T_co]):
    ...
    def __mul__(self, __value: SupportsIndex) -> tuple[_T_co, ...]: ...
```

Pyright is correctly apply the type rules here, so I don't consider this a bug.

It would be possible for pyright to ignore the typeshed type information and special-case this operation, but we generally don't do this. It's not scalable and creates maintenance problems as typeshed stubs are modified over time.

I'm going to close this for now. We could reconsider this decision in the future if there's sufficient signal from other pyright users (in the from of upvotes).
